test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +583,237 @@ contract('KyberNetworkProxy', function(accounts) {
+            let enable = await networkProxy.enabled();
+            assert(!enable);
+            await network.setEnable(true, { from: admin });
+            enable = await networkProxy.enabled();
+            assert(enable);
+        });
+
+        it("test maxGasPrice", async () => {
+            let expectedMaxGasPrice = await network.maxGasPrice();
+            let actualMaxGasPrice = await networkProxy.maxGasPrice();
+            Helper.assertEqual(expectedMaxGasPrice, actualMaxGasPrice);
+        });
+    });
+
+    describe("test events", async () => {
+        it("HintHandlerSet", async () => {
+            let newHintHandler = await MatchingEngine.new(admin);
+            let txResult = await networkProxy.setHintHandler(newHintHandler.address, { from: admin });
+            await expectEvent(txResult, "HintHandlerSet", {
+                hintHandler: newHintHandler.address
+            });
+            await networkProxy.setHintHandler(matchingEngine.address, { from: admin });
+        });
+
+        it("KyberNetworkSet", async () => {
+            let newNetWork = await KyberNetwork.new(admin);
+            let txResult = await networkProxy.setKyberNetwork(newNetWork.address, { from: admin });
+            await expectEvent(txResult, "KyberNetworkSet", {
+                newNetwork: newNetWork.address, oldNetwork: network.address,
+            })
+            await networkProxy.setKyberNetwork(network.address, { from: admin });
+        });
+
+        it("ExecuteTrade", async () => {
+            let srcToken = tokens[1]
+            let srcAmount = new BN(2).mul(new BN(10).pow(new BN(tokenDecimals[1])));
+            let destAddress = accounts[7];
+            let initialBalance = await Helper.getBalancePromise(destAddress);
+            let hint = web3.utils.fromAscii("PERM");
+            let fee = 5;
+
+            await srcToken.transfer(taker, srcAmount);
+            await srcToken.approve(networkProxy.address, srcAmount, { from: taker });
+            let rate = await networkProxy.getExpectedRateAfterFee(srcToken.address, ethAddress, srcAmount, fee, hint);
+            let txResult = await networkProxy.tradeWithHintAndFee(srcToken.address, srcAmount,
+                ethAddress, destAddress, maxDestAmt, calcMinRate(rate), platformWallet,
+                fee, hint, { from: taker }
+            );
+            let destBalanceAfter = await Helper.getBalancePromise(destAddress);
+            let destQty = new BN(destBalanceAfter).sub(new BN(initialBalance));
+            await expectEvent(txResult, "ExecuteTrade", {
+                trader: taker,
+                src: srcToken.address,
+                dest: ethAddress,
+                actualSrcAmount: srcAmount,
+                actualDestAmount: destQty,
+                platformWallet: platformWallet,
+                platformFeeBps: new BN(fee),
+            });
+        });
+    });
+
+    describe("test require condition", async () => {
```

Comment:
better name would be.
test reverting when using contract zero address.

Reply:
change as request